src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        User user = userFromDbOptional.get();
+        if (userFromDbOptional.isEmpty()
+                || !HashUtil.hashPassword(password, user.getSalt()).equals(user.getPassword())) {
+            throw new AuthenticationException("Can`t authenticate user. We don`t have this user.");
+        }
+        return user;
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
```

Comment:
You have bad logic here. Now I can register a new user even if the user has already been registered.
what this method should do: 
1. check if you have valid data in the variables
2. check if the user already exists, if not - register a new user